frontend/packages/operator-lifecycle-manager/src/components/create-operand.tsx

```diff
@@ -129,25 +142,174 @@ const fieldsForOpenAPI = (openAPI: SwaggerDefinition): OperandField[] => {
+                        ) as SpecCapability.fieldGroup,
+                        ...capabilitiesFor(thirdLevelVal.type),
+                      ],
+                    }),
+                  );
+                },
+              );
+            }
+
+            if (
+              _.values(val.properties).some((nestedVal) =>
+                _.values(nestedVal.properties).some((thirdLevelVal) =>
+                  _.values(thirdLevelVal.properties).every(
+                    (forthLevelVal) => !['object', 'array'].includes(forthLevelVal.type),
+                  ),
+                ),
+              )
+            ) {
+              return _.map(
+                val.properties,
+                (nestedVal, nestedKey: string): OperandField => {
+                  if (!['object', 'array'].includes(nestedVal.type)) {
+                    return {
+                      path: [key, nestedKey].join('.'),
+                      displayName: _.startCase(nestedKey),
+                      type: nestedVal.type,
+                      required: _.get(val, 'required', []).includes(nestedKey),
+                      validation: null,
+                      capabilities: [
+                        SpecCapability.internalFieldGroupForthSecLevel.concat(
+                          key,
+                        ) as SpecCapability.fieldGroup,
+                        ...capabilitiesFor(nestedVal.type),
+                      ],
+                    };
+                  }
+                  if (!_.has(nestedVal, 'properties')) {
+                    return null;
+                  }
+                  return _.map(
+                    nestedVal.properties || nestedVal.items.properties,
+                    (thirdLevelVal, thirdLevelKey: string): OperandField => {
+                      if (!['object', 'array'].includes(thirdLevelVal.type)) {
+                        return {
+                          path: [key, nestedKey, thirdLevelKey].join('.'),
+                          displayName: _.startCase(thirdLevelKey),
+                          type: thirdLevelVal.type,
+                          required: _.get(thirdLevelVal.properties, 'required', []).includes(thirdLevelKey),
+                          validation: null,
+                          capabilities: [
+                            SpecCapability.internalFieldGroupForthThirdLevel.concat(
+                              key,
+                            ) as SpecCapability.fieldGroup,
+                            SpecCapability.internalFieldGroupForthSecLevel.concat(
+                              nestedKey,
+                            ) as SpecCapability.fieldGroup,
+                            ...capabilitiesFor(thirdLevelVal.type),
+                          ],
+                        };
+                      }
+                      if (!_.has(thirdLevelVal, 'properties')) {
+                        return null;
+                      }
+                      return _.map(
```

Comment:
One initial observation, there is a lot of repeat of this block with just subtle differences.  I wonder if there is a way to factor out the shared bits into separate funcs.